server/src/main/java/com/google/musicanalysis/site/AnalysisServlet.java

```diff
@@ -14,30 +14,51 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.regex.Matcher; 
+import java.util.regex.Pattern; 
 
 @WebServlet("/api/analysis")
 public class AnalysisServlet extends HttpServlet {
   @Override
   protected void doGet(HttpServletRequest req, HttpServletResponse res)
       throws ServletException, IOException {
 
-    String videoName = req.getParameter("name");
-
-    assert videoName == null : "Something went wrong with sending to backend.";
+    String input = req.getParameter("name");
 
     // Use like this: {url_parameter, value}
     HashMap<String, String> videoArgs = new HashMap<>();
     HashMap<String, String> commentArgs = new HashMap<>();
     HashMap<String, String> nameArgs = new HashMap<>();
 
-    videoArgs.put("q", videoName);
-    videoArgs.put("type", "video");
-    String videoIdJson = new YoutubeRequest("search", videoArgs).getResult();
-    String videoId = getVideoId(videoIdJson);
+    String videoId = input; // assume user enters id
 
     commentArgs.put("part", "snippet");
-    commentArgs.put("videoId", videoId);
-    String commentsJson = new YoutubeRequest("commentThreads", commentArgs).getResult();
+    commentArgs.put("videoId", input);
+    String commentsJson;
+    
+    // if the input doesn't look like a standard id
+    if (thereIsWhiteSpace(input) || input.length() > 11) {
```

Comment:
Invert this logic. You want to check first if you can get a video by this "id". If it comes back with a 404, then you want to use the search API.

Reply:
You should also separate this logic out into helper functions now that this function is getting long. Something like:
```
Comments comments;
if (videoParam.length() == 11 && !thereIsWhitespace(videoParam)) {
  try {
    comments = getCommentsFromVideoId(videoParam);
  } catch {
    String videoId = getFirstVideoFromSearch(videoParam);
    comments = getCommentsFromVideoId(videoId);
  }
}
```